src/cplscheme/CouplingScheme.hpp

```diff
@@ -68,35 +68,29 @@ class CouplingScheme {
       double startTime,
       int    startTimeWindow) = 0;
 
-  /// Returns true, if initialize has been called.
-  virtual bool isInitialized() const = 0;
-
-  /**
-   * @brief Getter for _sendsInitializedData
-   * @returns _sendsInitializedData
-   */
-  virtual bool sendsInitializedData() const = 0;
-
   /**
-   * @brief Getter for _receivesInitializedData
-   * @returns _receivesInitializedData
+   * @brief Receives result of first advance, if this has to happen inside SolverInterface::initialize()
+   *
+   * This is only relevant for the second participant of the SerialCouplingScheme, because other coupling schemes only
+   * receive initial data in initialize. This part is implemented as a public function and has to be called explicitly
+   * from the outer scope (usually SolverInterfaceImpl), because the outer scope has to additional actions between
+   * CouplingScheme::initialize and CouplingScheme::receiveResultOfFirstAdvance
    */
-  virtual bool receivesInitializedData() const = 0;
+  virtual void receiveResultOfFirstAdvance(){
+      //@todo: Add security checks?
+  };
```

Comment:
No big fan of this function. We should carefully validate whether we need these actions between both calls.

Reply:
Done 52ae6c9e6ad74e84f7d47faef1289e094e4d1319